Photo-Editor/Photo-Editor/Controllers/PhotoController.swift

```diff
@@ -0,0 +1,187 @@
+//
+//  PhotoController.swift
+//  Photo-Editor
+//
+//  Created by Anthony Muñiz on 12/8/24.
+//
+
+import UIKit
+import Photos
+
+class PhotoController {
+    
+    func saveImageToDevice(image: UIImage, completion: @escaping (Error?) -> Void) {
+        PHPhotoLibrary.requestAuthorization { status in
+            if status == .authorized || status == .limited {
+                DispatchQueue.main.async {
```

Comment:
Swift Modern Concurrency was required for the capstone. You can replace DispatchQueue.main with await MainActor.run {